### DISCUSSION TITLE
Give Autopilot a different easing from Autoplay to make playing with it more enjoyable

### DISCUSSION THREAD

**Discussion by ramjam18** [ppy/osu#18798] [Ideas]
By default, Autoplay uses `Easing.Out` for eased movements. But because Autopilot inherits its aim from Autoplay, and `Easing.Out` moves away from hit objects relatively fast, this can result in more difficult gameplay when playing with Autopilot and not hitting in time.

For this reason, **I have actually made a small change in the code** that now changes the easing used in `OsuAutoGenerator.cs` based on if Autopilot is selected as a mod or not. **The code I made also already works as intended** and I get no compiler errors whatsoever.

**I would love to create a pull request for this** and offer my small contribution up for review, but to be honest, I have no idea how to do that, and GitHub's help page didn't exactly _help_ me understand.

If you have any thoughts or questions, feel free to leave them down below.

**EDIT: I've made a short video** that is supposed to show the difference between two easings I've tested against each other at 29ms late Relax gameplay. I've uploaded it [here](https://youtube.com/shorts/_hK0wCC0oxc). (Please excuse the poor quality... I don't have any good video editing tools.)

**Comment by bdach** [comment]
The problem with autopilot is already explained in a TODO in the class:

https://github.com/ppy/osu/blob/26541ad350622e345d9022d2a5fd11acd0d10b09/osu.Game.Rulesets.Osu/Mods/OsuModAutopilot.cs#L47-L48

To try and fix it by changing easing is missing the forest for the trees.

**Reply by ramjam18** [reply]
Yes, I am aware. My code doesn't address the core issue, but it is a slight upgrade for what is already there nonetheless.

In this case, consider my suggestion a temporary solution, as it seems the current implementation of Autopilot also is temporary.

**Reply by ramjam18** [reply]
Check out the video I've uploaded. As you can see, there is clearly a difference in playability, and so I think this change is worth considering.
